Frameworks/CoreText/DWriteWrapper.mm

```diff
@@ -0,0 +1,392 @@
+//******************************************************************************
+//
+// Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#include <COMIncludes.h>
+#import <wrl/implements.h>
+#include <COMIncludes_End.h>
+#import <mutex>
+#import <vector>
+#import <algorithm>
+#import <CoreFoundation/CFBase.h>
+#import "CoreTextInternal.h"
+#import "DWriteWrapper.h"
+
+using namespace std;
+using namespace Microsoft::WRL;
+
+static const wchar_t* TAG = L"_DWriteWrapper";
+static const wchar_t* c_defaultUserLanguage = L"en-us";
+static const wchar_t* c_defaultFontName = L"Gabriola";
+
+// TODO::
+// These should be removed once CTFont and UIFont are bridged.
+static IWLazyClassLookup s_lazyUIFont("UIFont");
+static const float c_defaultSystemFontSize = 15.0f;
+
+static ComPtr<IDWriteFactory> CreateDWriteFactoryInstance() {
+    ComPtr<IDWriteFactory> dwriteFactory;
+    THROW_IF_FAILED(DWriteCreateFactory(DWRITE_FACTORY_TYPE_SHARED, __uuidof(IDWriteFactory), &dwriteFactory));
+    return dwriteFactory;
+}
+
+static ComPtr<IDWriteFactory> GetDWriteFactoryInstance() {
+    static ComPtr<IDWriteFactory> s_dwriteFactory = CreateDWriteFactoryInstance();
+    return s_dwriteFactory;
+}
+
+/**
+ * Helper method to return the user set default locale string.
+ *
+ * @return use set locale string as wstring.
+ */
+static wstring _getUserDefaultLocaleName() {
+    wchar_t localeName[LOCALE_NAME_MAX_LENGTH];
+    int defaultLocaleSuccess = GetUserDefaultLocaleName(localeName, LOCALE_NAME_MAX_LENGTH);
```

Comment:
Yup I was right. Cannot use NSLocale because of the above reason

---
In reply to: [77869413](https://github.com/Microsoft/WinObjC/pull/950#discussion_r77869413) [](ancestors = 77869413,77756000)